### ISSUE TITLE
13.0.0-alpha.2 Animations issue

### ISSUE THREAD

**Issue by satanTime** [ike18t/ng-mocks#1586]
When switching from v12.5.1 to 13.0.0-alpha.2 (other setup unchanged: Angular 13.0.2, jasmine 3.10.1) I get the following two errors for some tests:

**Unexpected synthetic listener** 

This occurs when mocking a component that has animations defined. `NoopAnimationsModule` is imported into the testing module.

```
Error: Unexpected synthetic listener @showDismiss.done found. Please make sure that:
	  - Either `BrowserAnimationsModule` or `NoopAnimationsModule` are imported in your application.
	  - There is corresponding configuration for the animation named `@showDismiss.done` defined in the `animations` field of the `@Component` decorator (see https://angular.io/api/core/Component#animations).
	Error: Unexpected synthetic listener @showDismiss.done found. Please make sure that:
	  - Either `BrowserAnimationsModule` or `NoopAnimationsModule` are imported in your application.
	  - There is corresponding configuration for the animation named `@showDismiss.done` defined in the `animations` field of the `@Component` decorator (see https://angular.io/api/core/Component#animations).
	    at checkNoSyntheticProp (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/@angular/platform-browser/fesm2020/platform-browser.mjs:742:15)
	    at DefaultDomRenderer2.listen (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/@angular/platform-browser/fesm2020/platform-browser.mjs:732:24)
	    at BaseAnimationRenderer.listen (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/@angular/platform-browser/fesm2020/animations.mjs:299:30)
	    at listenerInternal (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2020/core.mjs:15245:44)
	    at ɵɵsyntheticHostListener (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2020/core.mjs:15155:5)
	    at Object.hostBindings (ng:///MockOfSbbNotification.js:18:21)
	    at invokeHostBindingsInCreationMode (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2020/core.mjs:10358:13)
	    at invokeDirectivesHostBindings (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2020/core.mjs:10341:17)
	    at createDirectivesInstances (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2020/core.mjs:9668:9)
	    at ɵɵelementStart (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2020/core.mjs:14864:9)
	Error: Expected undefined to be truthy.
	    at <Jasmine>
	    at UserContext.<anonymous> (http://127.0.0.1:9876/_karma_webpack_/webpack:/src/app/order/assignment/assignment-validation-notification/assignment-validation-notification.component.spec.ts:27:23)
	    at ZoneDelegate.invoke (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/zone.js/fesm2015/zone.js:372:1)
	    at ProxyZoneSpec.onInvoke (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/zone.js/fesm2015/zone-testing.js:287:1)
```

**passed into ng-mocks without Angular decorators**

This happens in two different tests:

1. Mocking the erronous component with `MockComponets(..)` and the component is annotated as a `@Component(..)` but inherits from a `@Directive()`.
2. The erronous component is _not_ mocked, but declared directly in `declarations` of the testing module - maybe it is because it is overwritten? (`.overrideComponent(WeightInputComponent , NG_CONTROL_TEST_PROVIDER)`)

```
Failed: WeightInputComponent declaration has been passed into ng-mocks without Angular decorators. Therefore, it cannot be properly handled. Highly likely,  ng-mocks is not in JIT mode. Otherwise, please create an issue on github: https://github.com/ike18t/ng-mocks/issues/new?title=False%20positive%20ng-mocks%20not%20in%20JIT. Thank you in advance for support.
	Error: WeightInputComponent declaration has been passed into ng-mocks without Angular decorators. Therefore, it cannot be properly handled. Highly likely,  ng-mocks is not in JIT mode. Otherwise, please create an issue on github: https://github.com/ike18t/ng-mocks/issues/new?title=False%20positive%20ng-mocks%20not%20in%20JIT. Thank you in advance for support.
	    at http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/ng-mocks/cjs/lib/common/error.missing-decorators.js:5:1
	    at push.67197.exports.default (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/ng-mocks/cjs/lib/common/core.reflect.body-catch.js:16:1)
	    at http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/ng-mocks/cjs/lib/common/core.reflect.directive-resolve.js:18:6
	    at decorateClass (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/ng-mocks/cjs/lib/mock-component/mock-component.js:276:1)
	    at MockComponent (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/ng-mocks/cjs/lib/mock-component/mock-component.js:310:1)
	    at Array.map (<anonymous>)
	    at MockComponents (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/ng-mocks/cjs/lib/mock-component/mock-component.js:286:1)
	    at UserContext.<anonymous> (http://127.0.0.1:9876/_karma_webpack_/webpack:/src/app/order/assignment/assignment-pos/assingment-pos-row-editor/assignment-pos-row-editor.component.spec.ts:43:23)
	    at ZoneDelegate.invoke (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/zone.js/fesm2015/zone.js:372:1)
	    at AsyncTestZoneSpec.onInvoke (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/zone.js/fesm2015/zone-testing.js:1196:1)
	NullInjectorError: R3InjectorError(DynamicTestModule)[WaggonNumberPipe -> WaggonNumberPipe]: 
	  NullInjectorError: No provider for WaggonNumberPipe!
	error properties: Object({ ngTempTokenPath: null, ngTokenPath: [ 'WaggonNumberPipe', 'WaggonNumberPipe' ] })
	NullInjectorError: R3InjectorError(DynamicTestModule)[WaggonNumberPipe -> WaggonNumberPipe]: 
	  NullInjectorError: No provider for WaggonNumberPipe!
	    at NullInjector.get (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2020/core.mjs:11139:27)
	    at R3Injector.get (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2020/core.mjs:11306:33)
	    at R3Injector.get (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2020/core.mjs:11306:33)
	    at NgModuleRef.get (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2020/core.mjs:22189:33)
	    at Object.get (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2020/core.mjs:21862:35)
	    at lookupTokenUsingModuleInjector (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2020/core.mjs:3349:39)
	    at getOrCreateInjectable (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2020/core.mjs:3461:12)
	    at ɵɵdirectiveInject (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2020/core.mjs:14720:12)
	    at NodeInjectorFactory.factory (ng:///AssignmentPosRowEditorComponent/ɵfac.js:4:59)
	    at getNodeInjectable (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/@angular/core/fesm2020/core.mjs:3556:44)
	TypeError: Cannot set properties of undefined (setting 'assignmentPos')
	    at UserContext.<anonymous> (http://127.0.0.1:9876/_karma_webpack_/webpack:/src/app/order/assignment/assignment-pos/assingment-pos-row-editor/assignment-pos-row-editor.component.spec.ts:153:30)
	    at ZoneDelegate.invoke (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/zone.js/fesm2015/zone.js:372:1)
	    at ProxyZoneSpec.onInvoke (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/zone.js/fesm2015/zone-testing.js:287:1)
	    at ZoneDelegate.invoke (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/zone.js/fesm2015/zone.js:371:1)
	    at Zone.run (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/zone.js/fesm2015/zone.js:134:1)
	    at runInTestZone (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/zone.js/fesm2015/zone-testing.js:567:1)
	    at UserContext.<anonymous> (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/zone.js/fesm2015/zone-testing.js:582:1)
	    at <Jasmine>
	TypeError: Cannot read properties of undefined (reading 'destroy')
	    at UserContext.<anonymous> (http://127.0.0.1:9876/_karma_webpack_/webpack:/src/app/order/assignment/assignment-pos/assingment-pos-row-editor/assignment-pos-row-editor.component.spec.ts:79:13)
	    at ZoneDelegate.invoke (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/zone.js/fesm2015/zone.js:372:1)
	    at ProxyZoneSpec.onInvoke (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/zone.js/fesm2015/zone-testing.js:287:1)
	    at ZoneDelegate.invoke (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/zone.js/fesm2015/zone.js:371:1)
	    at Zone.run (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/zone.js/fesm2015/zone.js:134:1)
	    at runInTestZone (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/zone.js/fesm2015/zone-testing.js:567:1)
	    at UserContext.<anonymous> (http://127.0.0.1:9876/_karma_webpack_/webpack:/node_modules/zone.js/fesm2015/zone-testing.js:582:1)
	    at <Jasmine>

```

_Originally posted by @daniel-sc in https://github.com/ike18t/ng-mocks/issues/1427#issuecomment-1011862415_

**Comment by satanTime** [comment]
Hi @daniel-sc, 

could you share how the declaration of the component or at least animation looks like and an example of failing test setup?

Thank you in advance!

**Comment by daniel-sc** [comment]
I'll try to provide some details tomorrow!

**Comment by daniel-sc** [comment]
Hi @satanTime 

it seems all issues are resolved for me with `13.0.0-alpha.6` :)

**Comment by satanTime** [comment]
Glad to hear! 

Thanks for the confirmation and happy coding!
